test/image/compare_pixels_test.js

```diff
@@ -48,37 +49,51 @@ var QUEUE_WAIT = 10;
  *      npm run test-image -- gl3d_* --queue
  */
 
-var pattern = process.argv[2];
-var mockList = getMockList(pattern);
-var isInQueue = (process.argv[3] === '--queue');
+var argv = minimist(process.argv.slice(2), {boolean: ['queue', 'filter' ]});
+var isInQueue = argv.queue;
+var filter = argv.filter;
 
-if(mockList.length === 0) {
-    throw new Error('No mocks found with pattern ' + pattern);
+var allMock = false;
+// If no pattern is provided, all mocks are compared
+if(argv._.length === 0) {
+    allMock = true;
+    argv._.push('');
 }
 
-// filter out untestable mocks if no pattern is specified
-if(!pattern) {
-    console.log('Filtering out untestable mocks:');
-    mockList = mockList.filter(untestableFilter);
-    console.log('\n');
-}
+// Build list of mocks to compare
+var allMockList = [];
+argv._.forEach(function(pattern) {
+    var mockList = getMockList(pattern);
 
-// gl2d have limited image-test support
-if(pattern === 'gl2d_*') {
-    if(!isInQueue) {
-        console.log('WARN: Running gl2d image tests in batch may lead to unwanted results\n');
+    if(mockList.length === 0) {
+        throw new Error('No mocks found with pattern ' + pattern);
     }
-    console.log('\nSorting gl2d mocks to avoid gl-shader conflicts');
-    sortGl2dMockList(mockList);
-    console.log('');
+
+    allMockList = allMockList.concat(mockList);
+});
+
+// To get rid of duplicates
+Array.prototype.unique = function() {
+    return this.filter(function(value, index, self) {
+        return self.indexOf(value) === index;
+    });
+};
+allMockList = allMockList.unique();
```

Comment:
Please rewrite this as:

```js
// To get rid of duplicates
function unique(value, index, self) {
    reurn self.indexOf(value) === index;
};
allMockList = allMockList.filter(unique);
```

adding things to the prototype of builtin objects is a (very) bad pattern.

Reply:
I agree! Sorry about that. Fixed in be336489a93f6600c6ac668f4f7b46b323d99e53